solr/core/src/java/org/apache/solr/search/facet/FacetProcessor.java

```diff
@@ -207,23 +208,22 @@ private void handleFilterExclusions() throws IOException {
     }
     if (excludeSet.size() == 0) return;
 
+    // recompute the base domain
+    fcontext.base = fcontext.searcher.getDocSet(getContextQueries().stream().filter(q -> !excludeSet.containsKey(q)).collect(Collectors.toList()));
+  }
+
+  protected List<Query> getContextQueries() {
     List<Query> qlist = new ArrayList<>();
 
     // TODO: somehow remove responsebuilder dependency
     ResponseBuilder rb = SolrRequestInfo.getRequestInfo().getResponseBuilder();
 
     // add the base query
-    if (!excludeSet.containsKey(rb.getQuery())) {
-      qlist.add(rb.getQuery());
-    }
+    qlist.add(rb.getQuery());
 
     // add the filters
     if (rb.getFilters() != null) {
-      for (Query q : rb.getFilters()) {
-        if (!excludeSet.containsKey(q)) {
-          qlist.add(q);
-        }
-      }
+      qlist.addAll(rb.getFilters());
```

Comment:
What's the reason for removing these excludeSet checks?

Reply:
They aren't removed. They are handled in a single pass: https://github.com/fullstorydev/lucene-solr/pull/127/files#diff-bf535b312bde0f43bcc717fd98de9d6bfb4b63c2706d94f43bda03995a199debR212. I can separate that line out of you want?

Reply:
Oh, 🤦 .  No need, LG!